services/question-service/entry-points/api/routes.ts

```diff
@@ -0,0 +1,63 @@
+import express from 'express';
+import * as questionUseCase from '../../domain/question-use-case';
+import { validateAddQuestionInput, validateUpdateQuestionInput } from './validators';
+
+export default function defineRoutes(expressApp: express.Application) {
+  const router = express.Router();
+
+  router.post('/',validateAddQuestionInput ,async (req, res, next) => {
+    try {
+      const addQuestionResponse = await questionUseCase.addQuestion(req.body);
+      return res.json(addQuestionResponse);
+    } catch (error) {
+      next(error);
+    }
+  });
+
+  router.get('/', async (_, res, next) => {
+    try {
+      const response = await questionUseCase.getAllQuestions();
+      res.json(response);
+    } catch (error) {
+      next(error);
+    }
+  })
```

Comment:
I think this route can take in query parameters as well, which are [often used to sort or filter resources](https://stackoverflow.com/questions/30967822/when-do-i-use-path-params-vs-query-params-in-a-restful-api). Then you can use the query params to filter the backend if it exists! I feel that this design would be more extensible in the sense that we don't have to create a new route every time we have a new type of query (i.e. filter questions by title, by difficulty, by category, etc.) as they would all be supported via this one route.

For filtering on Prisma, can take a look at [Prisma's filtering and sorting resource](https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting)

It could look something like this:

```suggestion
router.get('/', async (req, res, next) => {
  try {
    const queryParams: QueryParams = {
      title: req.query.title as string,
    };

    const response = await questionUseCase.getAllQuestions(queryParams);
    res.json(response);
  } catch (error) {
    next(error);
  }
});

```